File: demo/src/main/java/com/utcn/demo/entity/Answer.java
package com.utcn.demo.entity;


import jakarta.persistence.*;

@Entity
@Table(name = "answers")
public class Answer {

    @Id
    @Column(name="id")
    private Long answerId;

    @ManyToOne
    @JoinColumn(name="q_id",nullable = false)
    private Question answerQuestion;

    @OneToOne
    @JoinColumn(name="c_id",nullable = false)
    private Content answerContent;

    public Answer(){

    }

    public Answer(Long answerId, Question answerQuestion, Content answerContent) {
        this.answerId = answerId;
        this.answerQuestion = answerQuestion;
        this.answerContent = answerContent;
    }

    public Long getAnswerId() {
        return answerId;
    }

    public void setAnswerId(Long answerId) {
        this.answerId = answerId;
    }

    public Question getAnswerQuestion() {
        return answerQuestion;
    }

    public void setAnswerQuestion(Question answerQuestion) {
        this.answerQuestion = answerQuestion;
    }

    public Content getAnswerContent() {
        return answerContent;
    }

    public void setAnswerContent(Content answerContent) {
        this.answerContent = answerContent;
    }
}

File: app/pages/answer/answer.ts
import {Content} from "../content";
import {Question} from "../question/question";

export class Answer {

  private _answerId:number;
  private _answerQuestion:Question;
  private _answerContent:Content;


  constructor(answerId: number, answerQuestion: Question, answerContent: Content) {
    this._answerId = answerId;
    this._answerQuestion = answerQuestion;
    this._answerContent = answerContent;
  }


  get answerId(): number {
    return this._answerId;
  }

  set answerId(value: number) {
    this._answerId = value;
  }

  get answerQuestion(): Question {
    return this._answerQuestion;
  }

  set answerQuestion(value: Question) {
    this._answerQuestion = value;
  }

  get answerContent(): Content {
    return this._answerContent;
  }

  set answerContent(value: Content) {
    this._answerContent = value;
  }
}
